src/main/io/serial_cli.c

```diff
@@ -970,6 +974,82 @@ static void cliSerial(char *cmdline)
 
 }
 
+static void cliSerialPassthrough(char *cmdline)
+{
+    if (isEmpty(cmdline)) {
+        cliShowParseError();
+        return;
+    }
+
+    int id = -1;
+    uint32_t baud = 0;
+    baudRate_e baudIdx = BAUD_AUTO;
+    unsigned mode = 0;
+    char* tok = strtok(cmdline, " ");
+    int index = 0;
+
+    while (tok != NULL) {
+        switch(index) {
+            case 0:
+                id = atoi(tok);
+                break;
+            case 1:
+                baud = atoi(tok);
+                baudIdx = lookupBaudRateIndex(baud);
+                if (baudIdx == BAUD_AUTO) {
+                    printf("Parse error: invalid baud rate %d\r\n", baud);
+                    return;
+                }
+                break;
+            case 2:
+                if (strcasestr(tok, "rx"))
+                    mode |= MODE_RX;
+                if (strcasestr(tok, "tx"))
+                    mode |= MODE_TX;
+                break;
+        }
+        index++;
+        tok = strtok(NULL, " ");
+    }
+
+    serialPort_t *passThroughPort;
+    serialPortUsage_t *passThroughPortUsage = findSerialPortUsageByIdentifier(id);
+    if (!passThroughPortUsage || passThroughPortUsage->serialPort == NULL) {
+        if (!baud) {
+            printf("Port %d is not open, you must specify baud\r\n", id);
+            return;
+        }
+        if (!mode)
+            mode = MODE_RXTX;
+
+        passThroughPort = openSerialPort(id, FUNCTION_PASSTHROUGH, NULL,
+                                         baudIdx, MODE_RXTX,
+                                         SERIAL_NOT_INVERTED);
+        if (!passThroughPort) {
+            printf("Port %d could not be opened\r\n", id);
+            return;
+        }
+        printf("Port %d opened, baud=%d\r\n", id, baud);
+    } else {
+        passThroughPort = passThroughPortUsage->serialPort;
+        // If the user supplied a mode, override the port's mode, otherwise
+        // leave the mode unchanged. serialPassthrough() handles one-way ports.
+        printf("Port %d already open\r\n", id);
+        if (mode && !(passThroughPort->mode & mode)) {
```

Comment:
This won't work as intended ( consider `MODE_TX & MODE_RXTX` ).

Either allow changing mode completely `(passThroughPort->mode != mode)`  (and use `mode` in serialSetMode)
or allow enabling other direction `(mode & ~passThroughPort->mode)`